route/sport-router.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const Router = require('express').Router;
+const Sport = require('../model/sport.js');
+const jsonParser = require('body-parser').json();
+const debug = require('debug')('sportapp:sport-router');
+const createError = require('http-errors');
+
+const sportRouter = module.exports = new Router();
+
+sportRouter.post('/api/sports', jsonParser, function(req, res, next){
+  debug('POST /api/sports');
+  new Sport(req.body).save()
+  .then(sport => res.json(sport))
+  .catch(next);
+});
+
+sportRouter.get('/api/sports/:id', function(req, res, next){
+  debug('GET /api/sports/:id');
+  Sport.findById(req.params.id)
+  .then(sport => res.json(sport))
+  .catch(err => next(createError(404, err.message)));
+});
+
+sportRouter.delete('/api/sports/:id', function(req, res, next){
+  debug('DELETE /ap/sports/:id');
+  Sport.findByIdAndRemove(req.params.id)
+  .then(sport =>  {
+    res.statusCode = 204;
+    res.json(sport);
```

Comment:
No need to ``res.json()`` here as you're not actually sending anything with a delete method. Status 204 suffices. :)